### PR TITLE
Heading toolbar

### PR DESCRIPTION
**Pull request by SergioEstevao**
This PR adds support for the toolbar specific items to change the type of Header (h2, h3, h4).

screen shot 2018-09-03 at 12 44 37

### CODE REVIEW

File: src/block-management/block-manager.js [hunk 1]
```diff
@@ -164,14 +164,15 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	}
 
 	onChange( clientId: string, attributes: mixed ) {
-		// Update datasource UI
+		// Update Redux store
+		this.props.onChange( clientId, attributes );
```
Comment (SergioEstevao) [inline src/block-management/block-manager.js]: I pulled the onChange/redux change to before  to make act and mark the refresh variable before we go and change the block ourselfs bellow.

File: src/block-management/block-manager.js [hunk 2]
```diff
@@ -164,14 +164,15 @@ export default class BlockManager extends React.Component<PropsType, StateType>
...
+
+		// Change the data source
 		const index = this.getDataSourceIndexFromClientId( clientId );
 		const dataSource = this.state.dataSource;
-		const block = dataSource.get( index );
+		const block = { ...( dataSource.get( index ) ) };
```
Comment (SergioEstevao) [inline src/block-management/block-manager.js]: Ah sorry, I was trying to do another change here to see if I could just do a copy of the block and avoid conflicts with the reducer. Let me change that back.

**Comment by hypest** [comment]
Perhaps minor, I'm not seeing a toolbar button for the `h1` case as mentioned in the PR description @SergioEstevao. Should there be one?

**Comment by SergioEstevao** [comment]
It looks it's not part of the options that Gutenberg uses, I will update the ticket description.

**Pull request merged**